tools/ExecuteCheckOutputFilesAndClean.sh

```diff
@@ -0,0 +1,14 @@
+#!/bin/sh
+
+# Distributed under the MIT License.
+# See LICENSE.txt for details.
+
+@Python_EXECUTABLE@ @CMAKE_SOURCE_DIR@/tools/CleanOutput.py -v --force \
+                    --input-file $2 --output-dir @CMAKE_BINARY_DIR@/$3
+mkdir @CMAKE_BINARY_DIR@/$3
+cd @CMAKE_BINARY_DIR@/$3
+@CMAKE_BINARY_DIR@/bin/$1 --input-file $2 &&
+    @Python_EXECUTABLE@ @CMAKE_SOURCE_DIR@/tools/CheckOutputFiles.py \
+                        --input-file $2 --run-directory @CMAKE_BINARY_DIR@/$3
+@Python_EXECUTABLE@ @CMAKE_SOURCE_DIR@/tools/CleanOutput.py -v \
```

Comment:
The `ctest` currently doesn't fail when this `unittest` does. I think an easy way to facilitate that would be to chain lines 12 and 13 with `&&`. That means the output won't be removed when the test fails, but maybe that's actually good.